$OpenBSD: patch-test_openssl_test_ssl_rb,v 1.7 2022/02/01 18:54:28 tb Exp $

Remove a pend that hides missing support for SSL_CTX_set_client_cert_cb()
in the TLSv1.3 stack.

LibreSSL chose to name TLSv1.3 cipher suites consistently with previous names
while OpenSSL renamed their cipher suites in accordance with the RFCs. Adjust
test expectation to cope with this.

Index: test/openssl/test_ssl.rb
--- test/openssl/test_ssl.rb.orig
+++ test/openssl/test_ssl.rb
@@ -352,8 +352,9 @@ class OpenSSL::TestSSL < OpenSSL::SSLTestCase
   end
 
   def test_client_ca
-    pend "LibreSSL 3.2 has broken client CA support" if libressl?(3, 2, 0)
-
+    if ENV['SKIP_EXPECTED_FAILURES']
+      pend "skip failing client CA test"
+    end
     ctx_proc = Proc.new do |ctx|
       ctx.client_ca = [@ca_cert]
     end
@@ -1575,7 +1576,7 @@ class OpenSSL::TestSSL < OpenSSL::SSLTestCase
       pend 'TLS 1.3 not supported'
     end
 
-    csuite = ['TLS_AES_128_GCM_SHA256', 'TLSv1.3', 128, 128]
+    csuite = ['AEAD-AES128-GCM-SHA256', 'TLSv1.3', 128, 128]
     inputs = [csuite[0], [csuite[0]], [csuite]]
 
     start_server do |port|
